src/lms-harmonizer/edfi_lms_harmonizer/helpers/argparser.py

```diff
@@ -62,12 +73,23 @@ def build_mssql_adapter_with_integrated_security(self) -> None:
             self.trust_certificate,
         )
 
-    def build_postgresql_adapter(self, username: str, password: str) -> None:
+    def build_postgresql_adapter(self) -> None:
+        self._assert_credentials(self.username, self.password)
+
         self.adapter = sql_adapter.create_postgresql_adapter(
-            username, password, self.server, self.db_name, self._get_pgsql_server_port()
+            self.username, self.password, self.server, self.db_name, self._get_pgsql_server_port()
         )
 
     def get_adapter(self) -> sql_adapter.Adapter:
+        if not hasattr(self, 'adapter'):
```

Comment:
This is an interesting side effect of having a dataclass.  It throws an error _when reading an unset attribute_ that is declared in `__post_init__`. Workaround: `hasattr`. Note that we're still able to _set_ the attribute, thanks to `__post_init__`.